src/charm.py

```diff
@@ -233,6 +235,45 @@ def _on_restart_action(self, event):
         container.restart(self.name)
         self.set_active_unit_status()
 
+    @log_event_handler(logger)
+    def _on_update_status(self, event):
+        """Handle `update-status` events.
+
+        Args:
+            event: The `update-status` event triggered at intervals.
+        """
+        try:
+            self._validate()
+        except ValueError:
+            return
+
+        container = self.unit.get_container(self.name)
+        valid_pebble_plan = self._validate_pebble_plan(container)
+        if not valid_pebble_plan:
+            self._update(event)
+            return
+
+        check = container.get_check("up")
+        if check.status != CheckStatus.UP:
+            self.unit.status = MaintenanceStatus("Status check: DOWN")
+            return
+
+        self.set_active_unit_status()
+        if self.unit.is_leader():
+            self.ui._provide_server_status()
+
+    def _validate_pebble_plan(self, container):
+        """Validate Temporal server pebble plan.
+
+        Args:
+            container: application container
+
+        Returns:
+            bool of pebble plan validity
+        """
+        plan = container.get_plan().to_dict()
```

Comment:
Looking at the source for `get_plan` should we wrap this in a try-catch because I see the inner function can raise an exception when communicating with pebble.

Reply:
Good point, i've added the try-catch block.